src/assets/TokensController.ts

```diff
@@ -268,6 +295,68 @@ export class TokensController extends BaseController<
     }
   }
 
+  /**
+   * Adds isERC721 field to token object
+   * (Called when a user attempts to add tokens that were previously added which do not yet had isERC721 field)
+   *
+   * @param {string} tokenAddress - The contract address of the token requiring the isERC721 field added.
+   * @returns {Promise<object>} The new token object with the added isERC721 field.
+   *
+   */
+  async updateTokenType(tokenAddress: string) {
+    const { tokens } = this.state;
+    const tokenIndex = tokens.findIndex((token) => {
+      return token.address === tokenAddress;
+    });
+    tokens[tokenIndex].isERC721 = await this._detectIsERC721(tokenAddress);
+    this.update({ tokens });
+    return Promise.resolve(tokens[tokenIndex]);
```

Comment:
Nit: This Promise wrapper is redundant; the result will automatically be a Promise because this function is `async`.
```suggestion
    return Promise.resolve(tokens[tokenIndex]);
```